src/cobra/core/reaction.py

```diff
@@ -908,16 +916,13 @@ def copy(self) -> "Reaction":
         self._model = None
         for i in self._metabolites:
             i._model = None
-        for i in self._genes:
-            i._model = None
         # now we can copy
         new_reaction = deepcopy(self)
+        new_reaction.gpr = self.gpr.copy()
```

Comment:
Not necessary. The deepcopy above already guarantees that `Reaction.gpr` was copied as well. This even applies to a normal `copy()` with the PR that was merged before.

Also I think the slightly more pythonic way for those things is to overwrite `Object.__copy__` instead of adding a `Object.copy` function.

Reply:
assigning reaction.gpr to a value also forces update_genes_from_gpr(). I can make it explicit, by having the copy happen in the attributes, and then forcing update_genes_from_gpr() manually.

Reply:
Okay got it. Doing it explicitly seems to make more sense than having it as a side effect of the assignment.